docs/odcs.rst

```diff
@@ -24,84 +24,188 @@ idea is to not require the user to provide a yum repo url when building
+enforced when building layered images. For instance, consider the case of two
+images, X and Y. Y uses X as its parent image (FROM X). If image X was built
+with "beta" intent, image Y's intent can only be "beta" or "unsigned". If the
+dist-git repo for image Y has it configured to use "release" intent, this value
+will be downgraded to "beta" at build time.
+
+In the case where multiple composes are used, the least restrictive intent is
+used. Continuing with our previous signing intent example, let's say a container
+image build request uses two composes. Compose 1 was generated with no signing
+keys provided, and compose 2 was generated with "my-release-key". In this case,
+the intent is "unsigned".
+
+Because compose IDs can be passed in to OSBS, a mechanism will be in place to
+classify the intent of an existing compose. This will be done by inspecting the
+signing keys used for generating the compose and performing a reverse mapping to
+determine the intent. If a match cannot be determined, intent will be set to
+``None``, which is less restrictive than any defined signing intents.
+
+The ``signing_intent`` specified in ``container.yaml`` can be overridden with
+the build parameter of same name. This particular parameter will be ignored for
+autorebuilds. The value in ``container.yaml`` should always be used in that
+case. Note that the signing intent used by the compose of parent image is still
+enforced.
+
+Enabling ODCS Integration
+"""""""""""""""""""""""""
+
+The first step in enabling ODCS integration is by configuring the OSBS
+environment to use a specific ODCS server via the ``odcs_*`` configuration
+parameters.
+
+Next, ``container.yaml`` must be created and it must include the *compose*
+section. See example further down. This file must then be committed and pushed
+to git repository. Simply building the container image as usual will trigger
+ODCS interaction.
+
+
+ODCS Compose Checks
+"""""""""""""""""""
+
+When an ODCS compose ID is passed in as an argument, OSBS will perform some
+checks to ensure the compose is usable.
+
+- State done: verify the compose is not about to expire, extending if needed.
+  Continue with OSBS build once compose is in this state.
+
+- State failed: not supported, causes failure of the OSBS build.
+
+- State removed: request compose to be regenerated. In the future, we may want
+  to check the ``pre-remove`` state: https://pagure.io/odcs/issue/90
+
+- State wait|generating: wait for a certain amount of time (configurable in OSBS
+  environment) for compose request to move out of this state. Polling is
+  performed during wait period.
+
 API Changes
 -----------
 
-The following parameters will be used by ``buildContainer`` task defined by
-``koji-containerbuild`` plugin:
+- ``signing_intent <str, default None>``
+
+  Exposed in `koji-containerbuild`_, and `osbs-client`_.
+
+  May not be used if ODCS integration is not enabled.
```

Comment:
Is that "May be ignored" or "Must not be used"?

Reply:
Thanks. Changed it to "Must not be used".